src/library-authoring/units/LibraryUnitBlocks.tsx

```diff
@@ -254,7 +258,19 @@ export const LibraryUnitBlocks = ({ preview }: LibraryUnitBlocksProps) => {
     }
   }, [orderMutator]);
 
-  useEffect(() => setOrderedBlocks(blocks || []), [blocks]);
+  useEffect(() => {
+    // Create new ids which are unique using index.
+    // This is required to support multiple components with same id under a unit.
+    const newBlocks = blocks?.map((block, idx) => {
+      const newBlock: LibraryBlockMetadataWithUniqueId = {
+        ...block,
+        id: `${block.id}----${idx}`,
+        originalId: block.id,
```

Comment:
This is working well, thank you for fixing this issue @navinkarkera !
nit: you could opt to add a `uniqueId` instead of `originalId` to minimise the diff.

Reply:
@pomegranited I tried it but dnd-kit needs id to be unique, else it would require some major refactor in draggable list and sortable item components.